broker/src/main/java/io/camunda/zeebe/broker/system/configuration/DataCfg.java

```diff
@@ -36,26 +32,47 @@ public final class DataCfg implements ConfigurationEntry {
 
   private int logIndexDensity = 100;
 
+  // diskUsageMonitoring and watermark configs are deprecated and replaced by DiskCfg
   private boolean diskUsageMonitoringEnabled = DEFAULT_DISK_USAGE_MONITORING_ENABLED;
   private double diskUsageReplicationWatermark = DEFAULT_DISK_USAGE_REPLICATION_WATERMARK;
   private double diskUsageCommandWatermark = DEFAULT_DISK_USAGE_COMMAND_WATERMARK;
   private Duration diskUsageMonitoringInterval = DEFAULT_DISK_USAGE_MONITORING_DELAY;
-
+  private DiskCfg disk = new DiskCfg();
   private BackupStoreCfg backup = new BackupStoreCfg();
 
   @Override
   public void init(final BrokerCfg globalConfig, final String brokerBase) {
     directory = ConfigurationUtil.toAbsolutePath(directory, brokerBase);
 
-    if (!diskUsageMonitoringEnabled) {
-      LOG.info(
-          "Disk usage watermarks are disabled, setting all watermarks to {}",
-          DISABLED_DISK_USAGE_WATERMARK);
-      diskUsageReplicationWatermark = DISABLED_DISK_USAGE_WATERMARK;
-      diskUsageCommandWatermark = DISABLED_DISK_USAGE_WATERMARK;
-    }
-
     backup.init(globalConfig, brokerBase);
+
+    overrideDiskConfig();
+    disk.init(globalConfig, brokerBase);
+  }
+
+  private void overrideDiskConfig() {
+    // For backward compatibility, if the old disk watermarks are configured use those values
+    // instead of the new ones
+    if (diskUsageMonitoringEnabled != DEFAULT_DISK_USAGE_MONITORING_ENABLED) {
```

Comment:
:x: I'm not sure if we can solve this but I think to a user it is surprising (and a breaking change) that, when they **explicitly** set the default values, we discard them and use the new default values instead. I think we'd need a way to distinguish between users not setting this config and them explicitly setting default values. 
Maybe it's enough to just not assign default values to the old config, leaving them as null. And then only use new config if the old config has a null value.

Reply:
You are right that it is not covering all cases. I was thinking that it is not worth making it too complex, because the current default values are not good enough. So most likely users would have overwritten it. I will see if we can handle it better and handle all cases correctly.